workloads/server.hbs.md

```diff
@@ -0,0 +1,98 @@
+[using Service Toolkit claims](../getting-started/consume-services.md#stk-bind)
+is a good match for the `server` workload type.
+This is because it runs continuously to extract information from a RabbitMQ queue,
+and stores the resulting data locally in-memory and presents it through a web UI.
+
+If you have followed the Services Toolkit example, you can update the `spring-sensors-consumer-web`
+to use the `server` supply chain by changing the workload type by running:
+
+```console
+tanzu apps workload update spring-sensors-consumer-web --type=server
+```
+
+This shows the change in the workload label, and prompts you to accept the change.
+After the workload completes the new deployment, you'll notice a few differences:
+
+* The workload no longer advertises a URL. It's available within the cluster as
+`spring-sensors-consumer-web` within the namespace, but you must use
+`kubectl port-forward service/spring-sensors-consumer-web 8080` to access the web service on port 8080.
+
+    You can also set up a Kubernetes ingress rule to direct traffic from outside the cluster to the workload.
+    Using an ingress rule, you can specify that specific host names or paths must be routed to the application.
+    For more information about ingress rules, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/ingress/)
+
+* The workload no longer autoscales based on request traffic.
+  For the `spring-sensors-consumer-web` workload, this means that it never spawns
+a second instance that consumes part of the request queue.
+  Also, it does not scale down to zero instances.
+
+## `server`-specific Workload Parameters
+
+In addition to the common supply chain parameters, `server` workloads provide the ability to expose one or more network ports from the application to the Kubernetes cluster using the `ports` parameter. This parameter is a list of port objects, similar to a Kubernetes Service specification. If not set, the set of exposed ports will be determined by the applied container conventions in the cluster.
+
+The following configuration would expose two ports on the Kubernetes cluster under the `my-app` hostname:
+  * One service on port 25, which is redirected to port 2025 on the application
+  * One service on port 8080, which is routed to port 8080 on the application
+
+```yaml
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  name: my-app
+  labels:
+    apps.tanzu.vmware.com/workload-type: server
+spec:
+  params:
+  - name: ports
+    value:
+    - containerPort: 2025
+      name: smtp
+      port: 25
+    - port: 8080
+  ...
+```
+
+The `ports` parameter can be set from the `tanzu apps workload create` command line as `--param-yaml 'ports=[{"port": 8080}]'`.
+
+The following values are valid within the `ports` argument:
+
+| field | value |
+|-------|-------|
+| `port` | The port on which the application should be exposed to the rest of the cluster |
```

Comment:
Is it required to specify a port in a server workload? I assume it isn't required but calling it out just in case

Reply:
See line 66 -- conventions can fill these in of bit specified; any thoughts on how to represent this?

(There's a little mix of how-to and reference docs here}

Reply:
Ah, that didnt click the first time I read that. That makes sense though. Can the user get into a situation where their application isn't secure by default because they failed to define a port in the container conventions for their cluster? If so, we may want to add an example suggested config and/or a kubectl command that will show what ports are exposed.

Reply:
The port is exposed only within the cluster, and only for `server` type workloads.

The next defense would be to define NetworkPolicy to restrict within-cluster TCP access (because `server` workloads might or might not be HTTP).